bin/hdfcoinc/pycbc_plot_singles_timefreq

```diff
@@ -84,162 +82,187 @@ fig = pl.figure(figsize=(11,5.5))
 fig.subplots_adjust(left=0.06, right=0.95, bottom=0.09, top=0.95)
 ax = fig.gca()
 
-logging.info('Plotting strain spectrogram')
-Pxx, freq, t = mlab.specgram(strain, NFFT=1024, noverlap=1000,
-                             Fs=opts.sample_rate, mode='psd')
-del strain
-median_psd = np.median(Pxx, axis=1)
-median_psd_tile = np.tile(np.array([median_psd]).T, (1, len(t)))
-del median_psd
-Pxx /= median_psd_tile
-del median_psd_tile
-norm = LogNorm()
-pc = ax.pcolormesh(t + opts.gps_start_time - center_time, freq, Pxx,
-                   vmin=1, vmax=1000, norm=norm, cmap='afmhot_r')
-del freq, Pxx
-
-logging.info('Loading trigs')
-trig_f = HFile(opts.trig_file, 'r')
-trigs = trig_f[opts.detector]
-
-def rough_filter(snr, chisq, chisq_dof, end_time, tmp_id, tmp_dur):
-    return np.logical_and(end_time > opts.gps_start_time,
-                          end_time < opts.gps_end_time + tmp_dur)
-
-indices, snr, chisq, chisq_dof, end_time, template_ids, template_duration = \
+if center_time == -1.0:
+    title = '%s - no triggers' % opts.detector
+    ax.set_xlim(opts.gps_start_time - center_time, opts.gps_end_time - center_time)
+    ax.set_ylim(opts.f_low, opts.sample_rate / 2)
+    ax.set_yscale('log')
+    ax.grid(ls='solid', alpha=0.2)
+    ax.set_xlabel('Time - %.3f (s)' % center_time)
+    ax.set_ylabel('Frequency (Hz)')
+    ax.set_title(title)
+    caption = ("This plot shows the power spectrogram of the strain data, "
+               "normalized to its median over time, as a heatmap. The "
+               "time-frequency evolution of each single trigger is shown as a blue "
+               "curve. The light-blue curve is the loudest trigger by %s. Only the "
+               "loudest %d triggers by %s are shown. Green hatched areas denote "
+               "gated strain data. Red hatched areas denote vetoed time.") % \
+    (opts.rank, opts.num_loudest, opts.rank)
+    pycbc.results.save_fig_with_metadata(
+        fig, opts.output_file, cmd=' '.join(sys.argv),
+        title='Strain spectrogram and inspiral tracks for %s' % opts.detector,
+        caption=caption, fig_kwds={'dpi': 100})
+
+    logging.info('Done')
```

Comment:
For this code, I think it's easier to have:
```
if center_time == -1.0:
    commands_to_make_plot_as_you_have_it()
    sys.exit(0)
```
Then you don't have to indent the rest of the code in an `else` statement, and if we remove the `center_time == -1.0` thing it doesn't require a large unindent

Reply:
Done